tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/Support/OrderSummaryDataFactory.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Builders;
+using NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Utils;
+using NHSD.BuyingCatalouge.Ordering.Api.Testing.Data.Data;
+using NHSD.BuyingCatalouge.Ordering.Api.Testing.Data.EntityBuilder;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Steps.Support
+{
+    class OrderSummaryDataFactory
+    {
+        private readonly Settings _setting;
+
+        OrderSummaryDataFactory(Settings setting)
+        {
+            _setting = setting;
+        }
+
+        public async Task CreateData(string key, string orderId)
+        {
+            await DataFactory[key](orderId);
+        }
+
+        protected IDictionary<string, Func<string,Task>> DataFactory =>
+            new Dictionary<string, Func<string,Task>>()
+            {
+                {"complete", async (orderId) => await  OrderSummaryDataBuilder.CreateOrderSummaryData(orderId).Build().InsertAsync(_setting.ConnectionString)},
+                {"complete-with-1recipient-1associatedservice-fundingcomplete", async (orderId) => await  OrderSummaryDataBuilder.CreateOrderSummaryData(orderId)
+                    .RemoveAdditionalServices()
+                    .RemoveSolutionsServices()
+                    .Build()
+                    .InsertAsync(_setting.ConnectionString)
+                },
+                {"complete-with-0recipient-1associatedservice-fundingcomplete", async (string orderId) => await OrderSummaryDataBuilder.CreateOrderSummaryData(orderId)
```

Comment:
Please review the need to specify 'string'

Reply:
You requested that the Step not use a default default OrderId and it be specified and passed through. This is needed to pass the OrderId through to the builder?

Reply:
Just highlighting the difference between the line above.

`async (orderId) -> async (string orderId)`

"string" isn't defined on the first one, wondering why it is on the second entry. Yeah going to need to pass an order ID in.

Reply:
OK, sorry i misunderstood what you were saying